anomalib/data/base.py

```diff
@@ -4,33 +4,88 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import logging
+import warnings
 from abc import ABC, abstractmethod
+from pathlib import Path
 from typing import Dict, Optional, Tuple, Union
 
 import albumentations as A
-import cv2
 import numpy as np
+import pandas as pd
 from pandas import DataFrame
 from pytorch_lightning import LightningDataModule
+from pytorch_lightning.trainer.states import TrainerFn
 from pytorch_lightning.utilities.types import EVAL_DATALOADERS, TRAIN_DATALOADERS
 from torch import Tensor
 from torch.utils.data import DataLoader, Dataset
 
-from anomalib.data.utils import read_image
+from anomalib.data.utils import read_image, read_mask
 from anomalib.pre_processing import PreProcessor
 
 logger = logging.getLogger(__name__)
 
+TASK_CLASSIFICATION = "classification"
+TASK_SEGMENTATION = "segmentation"
+TASKS = (TASK_CLASSIFICATION, TASK_SEGMENTATION)
+
+SPLIT_TRAIN = "train"
+SPLIT_VAL = "val"
+SPLIT_TEST = "test"
+SPLITS = (SPLIT_TRAIN, SPLIT_VAL, SPLIT_TEST)
+
+LABEL_NORMAL = 0
+LABEL_ANOMALOUS = 1
+
+_SAMPLES_MANDATORY_COLUMNS_CLASSIFICATION = ["image_path", "label"]
+_SAMPLES_MANDATORY_COLUMNS_SEGMENTATION = _SAMPLES_MANDATORY_COLUMNS_CLASSIFICATION + ["mask_path"]
+
```

Comment:
Our style guide is based on google'. We should try to upload it somewhere. According to it, we should try to avoid global variables https://google.github.io/styleguide/pyguide.html#25-global-variables. I'd be in favour of having enums. Then we can have `Task.Classification` and `Task.Segmentation`.